packages/loader/container-loader/src/containerContext.ts

```diff
@@ -329,7 +329,7 @@ export class ContainerContext implements IContainerContext {
                 details: details ?? codeDetails,
             };
         } else {
-            return { module: loadCodeResult, details: codeDetails };
+            return loadCodeResult;
```

Comment:
Currently, this path handles the `IFluidModule` case (forcibly appending a details to it), while the above path handles the `IFluidModuleWithDetails`.  With this change, we don't expect this compat path to execute anymore (`ICodeDetailsLoader` should always result in an `IFluidModuleWithDetails`).

Although your change will result in compile-time breaks that will push people to update their code loaders, I think it might be possible that folks dynamically load their code loaders.  If that can happen, then retaining this runtime handling is probably good, plus instrument to see when that usage goes away.  Probably something like:

```suggestion
            // If "module" is not in the result, we are using a legacy ICodeLoader.  Fix the result up with details.
            // Once usage drops to 0 we can remove this compat path.
            this.taggedLogger.sendTelemetryEvent({ eventName: "LegacyCodeLoader" });
            return { module: loadCodeResult, details: codeDetails };
```